src/text/mod.rs

```diff
@@ -1,14 +1,173 @@
-//!       - Describe the difference between `TextStyle` and `MonoTextStyle`
-//!       - How is `Text::position` used by text renders dependent on `TextStyle::baseline`
-//!       - Add examples:
-//!         - Basic text without `TextStyle`
-//!         - Advanced text with `TextStyle`
-//!         - Draw text with multiple styles by using the result of `Text.draw(...)`
-//!       - Link to `renderer` module docs for users who want to implement custom renderers.
+//! Text drawing.
+//!
+//! The [`Text`] drawable can be used to draw text on a draw target. To construct a [`Text`] object
+//! at least a text string, position and character style are required. For advanced formatting
+//! options an additional [`TextStyle`] object might be required.
+//!
+//! Text rendering in embedded-graphics is designed to be extendable by text renderers for different
+//! font formats. To use a text renderer in an embedded-graphics project each renderer provides a
+//! character style object. This object is used to set the appearance of characters, like the text
+//! color or the used font. The available settings vary between different text renderer and are
+//! documented in the text renderer documentation.
+//!
+//! See the [`renderer` module] docs for more information about implementing custom text renderers.
+//!
+//! Embedded-graphics includes a text renderer for monospaced fonts in the [`mono_font`] module.
+//! Most examples will use this renderer and the associated [`MonoTextStyle`] character style.
+//! But they should be easily adaptable to any external renderer listed in the
+//! [external crates list].
+//!
+//! # Text style
+//!
+//! In addition to styling the individual characters the [`Text`] drawable also contains a
+//! [`TextStyle`] setting. The text style is used to set the alignment and line spacing of text
+//! objects.
+//!
+//! The [`alignment`] setting sets the horizontal alignment of the text. With the default value
+//! `Left` the text will be rendered to the right of the given text position. Analogously `Right`
+//! aligned text will be rendered to the left of the given position. `Center`ed text will extend
+//! equally to the left and right of the text position. If the text contains multiple lines each
+//! line will be aligned independently of the other lines.
+//!
+//! The [`baseline`] setting defines the vertical alignment of the first line of text. With the default
+//! setting of `Alphabetic` the glyphs will be drawn with their descenders below the given position.
+//! This means that the bottom of glyphs without descender (like 'A') will be on the same Y
+//! coordinate as the given position. The other baseline settings will position the glyphs relative
+//! to the EM box, without considering the baseline.
+//!
+//! If the text contains multiple lines only the first line will be vertically aligned based on the
+//! baseline setting. All following lines will be spaced according to the [`line_height`] setting.
```

Comment:
Maybe a small clarification that they're positioned relative to wherever the first line was rendered?

```suggestion
//! baseline setting. All following lines will be spaced relative to the first line, according to the [`line_height`] setting.
```